core/kernel/dt.c

```diff
@@ -473,6 +496,223 @@ void *get_secure_dt(void)
 }
 
 #if defined(CFG_EMBED_DTB)
+#ifdef CFG_DT_CACHED_NODE_INFO
+/* Reference to the FDT for which node information are cached */
+static const void *cached_node_info_fdt;
+
+/*
+ * struct cached_node - Cached information of a DT node
+ *
+ * @node_offset: Offset of the node in @cached_node_info_fdt
+ * @parent_offset: Offset of @node_offset parent node
+ * @address_cells: #address-cells property value of the parent node or 0
+ * @size_cells: #size-cells property value of the parent node or 0
+ * @phandle: Phandle associated to the node or 0 if none
+ */
+struct cached_node {
+	int node_offset;
+	int parent_offset;
+	int8_t address_cells;
+	int8_t size_cells;
+	uint32_t phandle;
+};
+
+/*
+ * struct dt_node_cache - Reference to cached information of DT nodes
+ * @array: Array of the cached node
+ * @count: Number of initialized cells in @array
+ * @alloced_array
+ */
+struct dt_node_cache {
+	struct cached_node *array;
+	size_t count;
+	size_t alloced_count;
+};
+
+static struct dt_node_cache dt_node_cache;
```

Comment:
How about allocating this and only keep a pointer to it so save memory once we're done with dt?

Reply:
Fair